rasterio/vfs.py

```diff
@@ -14,13 +16,17 @@ def parse_path(path, vfs=None):
     """Parse a file path or Apache VFS URL into its parts."""
     archive = scheme = None
     if vfs:
-        parts = vfs.split("://")
-        scheme = parts.pop(0) if parts else None
-        archive = parts.pop(0) if parts else None
+        parts = parse(vfs)
+        scheme = parts['scheme']
+        archive = parts['path']
+        if parts['authority'] and parts['authority'] != 'localhost':
+            archive = parts['authority'] + archive
     else:
-        parts = path.split("://")
-        path = parts.pop() if parts else None
-        scheme = parts.pop() if parts else None
+        parts = parse(path)
+        scheme = parts['scheme']
+        path = parts['path']
+        if parts['authority'] and parts['authority'] != 'localhost':
+            path = parts['authority'] + path
```

Comment:
@sgillies We use Google Cloud Platform at work, so I'll defer to your AWS knowledge.